git-remote-gosh/src/blockchain/tree/mod.rs

```diff
@@ -106,6 +103,29 @@ impl<'de> Visitor<'de> for GoshPathVisitor {
         formatter.write_str("expected valid gosh path string")
     }
 
+    fn visit_str<E>(self, v: &str) -> Result<Self::Value, E>
+    where
+        E: de::Error,
+    {
+        let mut gosh_path = GoshPath { inner: vec![] };
+        let mut cur = String::new();
+        for ch in v.chars() {
+            cur.push(ch);
+            // split by / but not \/
+            if cur.ends_with("/") && !cur.ends_with("\\/") {
+                cur.pop();
+                gosh_path
+                    .try_join(unescape_slashes(&cur))
+                    .map_err(E::custom)?;
+                cur = String::new();
+            }
+        }
+        gosh_path
+            .try_join(unescape_slashes(&cur))
+            .map_err(E::custom)?;
+        Ok(gosh_path)
+    }
+
     fn visit_string<E>(self, v: String) -> Result<Self::Value, E>
```

Comment:
Can you reuse visit_str then?

Reply:
👍

Reply:
resolved in https://github.com/gosh-sh/gosh/pull/76/commits/1de37329014f57da9f88e56811806d8851405a92